### PR TITLE
Add peephole for len(x.size())

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59051 Add peephole for len(x.size())**
* #58932 Add more shape functions for mobilenet
* #58890 Switch symbolic shape registry to operator map'
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* #56828 Add unary/binary ops necessary for mobilenet

Differential Revision: [D28727247](https://our.internmc.facebook.com/intern/diff/D28727247)

### CODE REVIEW

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727247).

File: test/jit/test_peephole.py [hunk 1]
```diff
@@ -529,3 +529,28 @@ def foo(x: int):
...
+    def test_peephole_len_list(self):
+        @torch.jit.script
+        def foo(x):
+            return len(x.size())
+
+        self.run_pass("peephole", foo.graph)
+        FileCheck().check("aten::len").run(foo.graph)
+        inputs = list(foo.graph.inputs())
+        inputs[0].setType(inputs[0].type().with_sizes([None, None]))
+        self.run_pass("peephole", foo.graph)
```
Comment (Krovatkin) [inline test/jit/test_peephole.py:542]: should `peephole` be running shape ops by default?

**Review by Krovatkin — APPROVED** [review]
:shipit:

File: test/jit/test_peephole.py (hunk 1 above)
Comment (eellison) [reply test/jit/test_peephole.py:542]: yes, it is only the weird pipeline in profiling_executor where we shouldnt

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727247).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28727247).

**Pull request closed without merging**